examples/image_reconstruction.ipynb

```diff
@@ -0,0 +1,810 @@
+    "valid_dataset = Subset(full_train_dataset, valid_indices)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "lXTbdOE1w_P1"
+   },
+   "source": [
+    "The downloaded MNIST dataset format is for classification, which means each sample contains an image and a label (the digit drawn in the image). However, for image reconstruction, the dataset should contain an input image and a target image, which are simply the same. Hence, using the code below, we define a new dataset that wraps an MNIST dataset and provides an image as an input and sets that image as its target. In other words, we change the format of each dataset sample from (image, label) to the (image, image)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "vWtcE1zIw_P2"
+   },
+   "outputs": [],
+   "source": [
+    "class ImageReconstructionDataset(Dataset):\n",
+    "\n",
+    "    def __init__(self, dataset):\n",
+    "        self.dataset = dataset\n",
+    "          \n",
+    "    def __getitem__(self, index):\n",
+    "        input_image = self.dataset[index][0]\n",
+    "        target_image = input_image  # In image reconstruction, input and target images are the same.\n",
+    "          \n",
+    "        return input_image, target_image\n",
+    "      \n",
+    "    def __len__(self):\n",
+    "        return len(self.dataset)   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "-DuRMsUVzXsr"
+   },
+   "source": [
+    "Finally, in the section below, we wrap the MNIST datasets into our wrapper and create data loaders for them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "T125CnoLw_P2"
+   },
+   "outputs": [],
+   "source": [
+    "train_dataset_new = ImageReconstructionDataset(train_dataset)\n",
+    "valid_dataset_new = ImageReconstructionDataset(valid_dataset)\n",
+    "test_dataset_new = ImageReconstructionDataset(test_dataset)\n",
+    "\n",
+    "train_dataloader = DataLoader(train_dataset_new, batch_size=batch_size, shuffle=True)\n",
+    "valid_dataloader = DataLoader(valid_dataset_new, batch_size=batch_size, shuffle=False)\n",
+    "test_dataloader = DataLoader(test_dataset_new, batch_size=1, shuffle=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "M4oNvCAFzrSV"
+   },
+   "source": [
+    "## Convolutional Autoencoder"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The most frequently used network for image reconstruction is the Autoencoder. In this section, we are going to define our own autoencoder. The encoder section tries to encode the input image and consequently, the decoder tries to decode the features and reconstruct the original image. As our input dataset (MNIST) contains images with low resolution and low complexity, we preferred not to design a complex network, in order to avoid overfitting."
```

Comment:
```suggestion
    "The most frequently used network for image reconstruction is the autoencoder. In this section, we are going to define our own autoencoder. The encoder section tries to encode the input image into features and consequently, the decoder tries to decode the features and reconstruct the original image. As our input dataset (MNIST) contains images with low resolution and low complexity, we preferred not to design a complex network in order to avoid overfitting."
```